### ISSUE TITLE
Compatible with Higher CUDA Versions?

### ISSUE THREAD

**Issue by ldkong1205** [xing0047/TPS#1]
Hi @xing0047 , thanks for open-sourcing your work!

Is the current TPS codebase compatible with higher CUDA versions (for newer PyTorch versions), for example, 11.3?

The current configuration requires torch==1.2.0 torchvision==0.4.0, which is only viable for lower CUDA versions like CUDA 10.0 and 9.2.

I have tried to compile the resample2d package under higher torch versions but it is not successful. It seems highly dependent on torch==1.2.0 torchvision==0.4.0.

Looking forward to your help.

**Comment by YangLi309** [comment]
I tried with higher pytorch and torchvision version (i.e., torch 1.11.0, torchvision 0.12.0, cuda 11.3). The only modification is to change the sample2d's setup.py a bit: change c++11 to c++14. All good on my 3090 GPU.

**Comment by ldkong1205** [comment]
> I tried with higher pytorch and torchvision version (i.e., torch 1.11.0, torchvision 0.12.0, cuda 11.3). The only modification is to change the sample2d's setup.py a bit: change c++11 to c++14. All good on my 3090 GPU.

Hi @YangLiYL1994 , thank you for your reply! I will try it.

By the way, have you tested the results on higher PyTorch and TorchVision versions? The source-only results from my side (/w torch 1.11.0) are much higher than the reported results in the original papers (TPS and DA-VSN).

**Comment by YangLi309** [comment]
@ldkong1205 I haven't done it yet, I just set up the environment yesterday, and only ran experiments of viper->cityscapes. I will double-check it and let you know my results.  By the way, do you know what's the format of the dataset Synthia-Seq? I downloaded the data from http://synthia-dataset.net/download/716/ (SYNTHIA-SEQS-04-DAWN), however, the format is different from the one provided in authors' dataloader. The downloaded one's structure is SEQS-04-DAWN/rgb/Stereo_Right(Left)/Omni_B(F/L/R)/xxx.png, but the data dir structure in TPS seems to be SEQS-04-DAWN/rgb/xxx.png. How did you solve this?

**Comment by ldkong1205** [comment]
> @ldkong1205 I haven't done it yet, I just set up the environment yesterday, and only ran experiments of viper->cityscapes. I will double-check it and let you know my results. By the way, do you know what's the format of the dataset Synthia-Seq? I downloaded the data from http://synthia-dataset.net/download/716/ (SYNTHIA-SEQS-04-DAWN), however, the format is different from the one provided in authors' dataloader. The downloaded one's structure is SEQS-04-DAWN/rgb/Stereo_Right(Left)/Omni_B(F/L/R)/xxx.png, but the data dir structure in TPS seems to be SEQS-04-DAWN/rgb/xxx.png. How did you solve this?

Hi @YangLiYL1994 , please refer to this issue: https://github.com/Dayan-Guan/DA-VSN/issues/3

**Comment by YangLi309** [comment]
Sweet! Thank :)

**Comment by YangLi309** [comment]
Hi @ldkong1205, I finally have got all results on both datasets, please have a look:
Source&target
Viper2city
best mIoU: 47.92
[80.2, 32.2, 82.8, 9.7, 28.7, 27.6, 81.2, 35.4, 79.7, 62.4, 81.6, 40.5, 51.0, 21.1, 4.6]

Syn2city
Current best mIoU: 52.58
[87.7, 48.3, 73.4, 24.3, 20.8, 29.0, 71.7, 68.5, 56.2, 13.2, 85.3]

Source only
Viper2city
Current best mIoU: 36.72
[58.7, 17.4, 75.1, 8.9, 28.7, 14.2, 82.7, 9.0, 75.5, 60.0, 60.8, 5.6, 23.3, 15.7, 15.4]

Syn2city
Current best mIoU: 39.62
[62.8, 23.6, 74.2, 26.5, 7.9, 22.7, 67.6, 57.8, 41.2, 17.6, 33.8]

the source only results look ok, which are not too much different from the reported ones.

Test env: cuda 11.3, pytorch 1.11.0

However, my concern is the model performance of some classes is way lower than reported. Think the pseudo labelling technique will cause such instability issue.

**Comment by ldkong1205** [comment]
> Hi @ldkong1205, I finally have got all results on both datasets, please have a look:
> 
> Source&target
> 
> Viper2city
> 
> best mIoU: 47.92
> 
> [80.2, 32.2, 82.8, 9.7, 28.7, 27.6, 81.2, 35.4, 79.7, 62.4, 81.6, 40.5, 51.0, 21.1, 4.6]
> 
> 
> 
> Syn2city
> 
> Current best mIoU: 52.58
> 
> [87.7, 48.3, 73.4, 24.3, 20.8, 29.0, 71.7, 68.5, 56.2, 13.2, 85.3]
> 
> 
> 
> Source only
> 
> Viper2city
> 
> Current best mIoU: 36.72
> 
> [58.7, 17.4, 75.1, 8.9, 28.7, 14.2, 82.7, 9.0, 75.5, 60.0, 60.8, 5.6, 23.3, 15.7, 15.4]
> 
> 
> 
> Syn2city
> 
> Current best mIoU: 39.62
> 
> [62.8, 23.6, 74.2, 26.5, 7.9, 22.7, 67.6, 57.8, 41.2, 17.6, 33.8]
> 
> 
> 
> the source only results look ok, which are not too much different from the reported ones.
> 
> 
> 
> Test env: cuda 11.3, pytorch 1.11.0
> 
> 
> 
> However, my concern is the model performance of some classes is way lower than reported. Think the pseudo labelling technique will cause such instability issue.

Hi @YangLiYL1994 , thank you very much for your updates! For the experiments, are you using the default configurations given in the code? Specifically, is the batch size fixed as 1?

**Comment by YangLi309** [comment]
Yeah, that's right. I followed the default settings.

**Comment by ldkong1205** [comment]
> Yeah, that's right. I followed the default settings.

Got it. Would you mind to try larger batch size with larger learning rate (for example 1e-2) and smaller training steps? The results might change a bit, at least from my side. Thank you!

**Comment by YangLi309** [comment]
> > Yeah, that's right. I followed the default settings.
> 
> Got it. Would you mind to try larger batch size with larger learning rate (for example 1e-2) and smaller training steps? The results might change a bit, at least from my side. Thank you!

Yeah, I am doing it now :)

**Comment by ldkong1205** [comment]
> > > Yeah, that's right. I followed the default settings.
> > 
> > 
> > Got it. Would you mind to try larger batch size with larger learning rate (for example 1e-2) and smaller training steps? The results might change a bit, at least from my side. Thank you!
> 
> Yeah, I am doing it now :)

Hi @YangLiYL1994, any updates for this?
